src/java/org/apache/nutch/crawl/CrawlDb.java

```diff
@@ -261,30 +262,68 @@ public int run(String[] args) throws Exception {
       additionsAllowed = false;
     }
     
-    String crawldb = crawlId+"/crawldb";
-    String segment_dir = crawlId+"/segments";
-    File segmentsDir = new File(segment_dir);
-    File[] segmentsList = segmentsDir.listFiles();  
-    Arrays.sort(segmentsList, new Comparator<File>(){
-      @Override
-      public int compare(File f1, File f2) {
-        if(f1.lastModified()>f2.lastModified())
-          return -1;
-        else
-          return 0;
-      }      
-    });
+    Path crawlDb;
+    if(args.containsKey(Nutch.ARG_CRAWLDB)) {
+    	Object crawldbPath = args.get(Nutch.ARG_CRAWLDB);
+    	if(crawldbPath instanceof Path) {
+    		crawlDb = (Path) crawldbPath;
```

Comment:
no probs.
We can always sort it out pre-release. It can be a final ticket to format code.
Thank you for attention to detail.